text/0000-wallet-update-process.md

```diff
@@ -0,0 +1,170 @@
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Wallet Update Procedures
+
+The following sections outlines the processes used in the creation and update of outputs and related transactions within the wallet.
+
+### Transaction and Output Creation
+
+#### Coinbase Output + Transaction Creation
+
+Coinbase outputs and their related transactions are created via the wallet listener as part of the mining process. The workflow is:
+
+ * When mining, create a potential coinbase output for the target block.
+ * If the block is accepted and the output is detected in the UTXO set (via the [Update by Output process](#)), create a transaction log entry of type 'Confirmed Coinbase', with 'confirmed' set to true.
+
+#### Transacting (Payer -> Payee) Output + Transaction Creation
+
+Outputs and tranasctions are created during the transaction exchange process as follows. Note that the invoicing workflow is mostly identical with the roles of Payer and Payee reversed, so this workflow is not outlined separately.
+
+ * The sender creates a new 'blank' transaction Slate, adding inputs and change outputs to the slate.
+ * The sender generates a random kernel offset. (The kernel offset is always generated by the slate creator)
+ * The sender sends the slate to the payee (via file or http). If sending synchronously, (e.g via http)  the associated transaction is saved to the log after a response from the payee's listener. If sending asynchronously, (e.g. via file), the transaction is saved immediately. 
+ * When saved, the associated transaction is set to type 'Sent Tx'  with a status of 'unconfirmed', inputs are locked internally and a change output is added with status 'unconfirmed'.
+ * The payee receives the slate, creates (an) output(s) for the received amount with status unconfirmed, and immediately stores a transaction in their log of type 'Received Tx' with confirmed set to false. 
+ * The payee calculates and saves the transaction kernel for later reference.
+ * The slate is returned to the payer for completion, who calculates and saves the transaction kernel. The transaction is sent to a node for validation.
+
+### Transaction and Output Statue Update Processes
+
+#### Update by Output Process
+
+The Update by Output Process is the main method by which the wallet updates its internal state, decides whether a transaction has been confirmed and when to remove or confirm individual outputs. This process works as follows:
+
+* For every unspent output in the wallet:
+* If the output's status in the user's wallet is 'unconfirmed' and it is present in the UTXO set, change the output's status to 'confirmed', and update the associated transaction status to 'confirmed'.
+* If the output's status in the user's wallet is 'confirmed' and it no longer appears in the UTXO set, set it's status to 'Spent'. Note these outputs will usually be locked so they cannot be selected for spending again.
```

Comment:
Reorgs can technically add an output to the utxo set, remove it, and then later add it back. Unlikely scenario, but do you plan on handling that somehow to make sure additional transaction log entries don't get created? It's probably not an easy thing to do, so we don't need to handle it right away, I'm just curious what the plan is for this.

Reply:
Ah, you already answer this in the check-repair section.